### PR TITLE
SLIB-137 - Add support for Belgium person code

### PR DESCRIPTION
**Pull request by Johxzu**

### CODE REVIEW

File: src/main/java/ee/sk/smartid/rest/dao/SemanticsIdentifier.java
Comment (umuser) [inline src/main/java/ee/sk/smartid/rest/dao/SemanticsIdentifier.java:7]: Update license.

File: src/test/java/ee/sk/smartid/dao/SemanticsIdentifierTest.java [hunk 1]
```diff
@@ -57,4 +57,11 @@ public void constructor3() {
         assertThat(semanticsIdentifier.getIdentifier(), is("PNOLV-CCC-DDDDD"));
     }
 
+    @Test
+    public void constructor4() {
```
Comment (umuser) [inline src/test/java/ee/sk/smartid/dao/SemanticsIdentifierTest.java:61]: If you take a look in the SemanticsIdentifier it has 4 constructors, but the 4th constructor takes only a single string value as an input. Your example lines more with constructor3 test, but the order of constructors and test names do not match. 

All the tests should have more descriptive names. For example construct1 can be named to initializeWithStringValues and constructor2 can be named to initializeWithStringValuesAndIdentityType
or shorter initializeWithIdentityType.

With enum values you can also leverage @ParameterizedTest.
```java
    @ParameterizedTest
    @EnumSource(SemanticsIdentifier.IdentityType.class)
    public void initializeWithIdentityType(SemanticsIdentifier.IdentityType identityType) {
        SemanticsIdentifier semanticsIdentifier = new SemanticsIdentifier(identityType, "BB", "CCC");

        assertThat(semanticsIdentifier.getIdentifier(), is(identityType + "BB-CCC"));
    }
```

For multiple enum values and expected results look into @ArgumentsSource. 
In repo CertificateLevelArgumentProvider can be used as an example.

File: src/test/java/ee/sk/smartid/dao/SemanticsIdentifierTest.java
Comment (umuser) [inline src/test/java/ee/sk/smartid/dao/SemanticsIdentifierTest.java:7]: Update license

File: src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java
Comment (umuser) [inline src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java:7]: Update license

File: src/test/java/ee/sk/smartid/util/NationalIdentityNumberUtilTest.java [hunk 2]
```diff
@@ -120,6 +120,59 @@ public void parseLvDateOfBirth_invalidMonth_throwsException() {
         assertThat(unprocessableSmartIdResponseException.getMessage(), is("Unable get birthdate from Latvian personal code 131365-1234"));
     }
 
+    @Test
+    public void getDateOfBirthFromIdCode_belgianIdCode_returns() {
```
Comment (umuser) [inline src/test/java/ee/sk/smartid/util/NationalIdentityNumberUtilTest.java:124]: You could group BE related tests with [Nested tests](https://docs.junit.org/6.1.3/writing-tests/nested-tests.html).

File: src/test/java/ee/sk/smartid/util/NationalIdentityNumberUtilTest.java
Comment (umuser) [inline src/test/java/ee/sk/smartid/util/NationalIdentityNumberUtilTest.java:7]: Update license

File: src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java [hunk 3]
```diff
@@ -134,6 +141,47 @@ public static LocalDate parseLvDateOfBirth(String lvNationalIdentityNumber) {
...
+        if (BE_UNKNOWN_BIRTH_MONTH_AND_DAY.equals(birthMonthAndDay)) {
+            logger.debug("Person has a Belgian national register number that does not carry birthdate info");
+            return null;
+        }
+
+        String birthDateYyyyMmDd = determineBeBirthYear(beNationalIdentityNumber) + birthMonthAndDay;
+        try {
+            return LocalDate.parse(birthDateYyyyMmDd, DATE_FORMATTER_YYYY_MM_DD);
+        } catch (DateTimeParseException e) {
+            throw new UnprocessableSmartIdResponseException("Unable get birthdate from Belgian personal code " + beNationalIdentityNumber, e);
```
Comment (umuser) [inline src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java:172]: Should be "Unable to get ..."

File: src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java [hunk 4]
```diff
@@ -134,6 +141,47 @@ public static LocalDate parseLvDateOfBirth(String lvNationalIdentityNumber) {
...
+        }
+    }
+
+    private static String determineBeBirthYear(String beNationalIdentityNumber) {
+        long codeWithoutCheckDigits = Long.parseLong(beNationalIdentityNumber.substring(0, 9));
+        int checkDigits = Integer.parseInt(beNationalIdentityNumber.substring(9));
+        String birthYearTwoDigit = beNationalIdentityNumber.substring(0, 2);
+
+        boolean bornBefore2000 = BE_CHECK_DIGITS_MODULUS - (codeWithoutCheckDigits % BE_CHECK_DIGITS_MODULUS) == checkDigits;
+        return (bornBefore2000 ? "19" : "20") + birthYearTwoDigit;
```
Comment (umuser) [inline src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java:182]: There is also an algorithm to check if person was born in 20xx. And if that result does not match checkDigits then invalid person identifier was provided. I linked the algorithm description under the task.

**Review by umuser — CHANGES_REQUESTED** [review]

File: src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java [hunk 5]
```diff
@@ -134,6 +145,95 @@ public static LocalDate parseLvDateOfBirth(String lvNationalIdentityNumber) {
...
+     * but the check digit is always calculated over the number as issued.
+     * <p>
+     * If birth month and day are not known (both are zeroes) then null is returned.
+     *
+     * @param beNationalIdentityNumber Belgian national register number
+     * @return Date of birth or null if the personal code does not carry birthdate info
+     * @throws UnprocessableSmartIdResponseException if the national identity number is invalid or date cannot be parsed
+     */
+    public static LocalDate parseBeDateOfBirth(String beNationalIdentityNumber) {
+        if (beNationalIdentityNumber == null || !BE_PERSON_CODE_PATTERN.matcher(beNationalIdentityNumber).matches()) {
```
Comment (umuser) [inline src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java]: Image

Add test with null value.

Comment (umuser) [inline src/main/java/ee/sk/smartid/util/NationalIdentityNumberUtil.java]: Add test for null input value

File: CHANGELOG.md
Comment (umuser) [inline CHANGELOG.md]: This should have date of Mar 18, 2026. This version has been already published.

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,6 +7,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Changes
 
+- Added Belgium (`BE`) support:
+  - Added `BE` to `SemanticsIdentifier.CountryCode`.
+  - Added `NationalIdentityNumberUtil.parseBeDateOfBirth()` which extracts the date of birth from a Belgian national
+    register number. The century is deduced from the check digit; numbers whose check digit matches neither century
+    are rejected as invalid. Bisnummer, which carries the birth month increased by 20 or 40, is supported. Numbers
+    where the birth month and day are unknown (both zeroes) do not carry the date of birth and `null` is returned.
+  - `NationalIdentityNumberUtil.getDateOfBirth()` now also handles identities with country `BE`.
```
Comment (umuser) [inline CHANGELOG.md]: Your changes should be under new version 3.3.0

**Review by umuser — CHANGES_REQUESTED** [review]
